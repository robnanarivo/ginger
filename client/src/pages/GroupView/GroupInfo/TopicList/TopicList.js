import React from 'react';
import Box from '@mui/material/Box';
import Chip from '@mui/material/Chip';

function displayTopics(topics) {
  const ret = [];
  topics.forEach((topic) => {
    ret.push(
      <Chip
        label={topic}
        key={topic}
        sx={{
          color: '#017DD7',
          bgcolor: '#EFEFF1',
          marginLeft: '3%',
          marginTop: '3%',
        }}
      />,
    );
  });
  return ret;
}

export default function TopicList(props) {
  const { topics } = props;
  return (
    <Box
      sx={{
        display: 'flex',
        flexDirection: 'row',
        flexWrap: 'wrap',
        width: '100%',
      }}
    >
      {displayTopics(topics)}
    </Box>
  );
}
