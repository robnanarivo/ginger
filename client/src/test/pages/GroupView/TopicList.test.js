import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import TopicList from '../../../pages/GroupView/GroupInfo/TopicList';
import 'jest-styled-components';

describe('TopicList unit test', () => {
  it('TopicList Snapshot test', () => {
    const tree = renderer
      .create(<TopicList
        topics={['Sports', 'Universe', 'Science']}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('TopicList contains correct element', () => {
    render(<TopicList
      topics={['Sports', 'Universe', 'Science']}
    />);
    // contains Sports topic
    expect(screen.getByText('Sports')).toBeInTheDocument();
    // contains Universe topic
    expect(screen.getByText('Universe')).toBeInTheDocument();
  });
});
