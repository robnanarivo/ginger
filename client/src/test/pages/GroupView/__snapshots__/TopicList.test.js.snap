// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicList unit test TopicList Snapshot test 1`] = `
<div
  className="MuiBox-root css-1290pys"
>
  <div
    className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1dzpuen-MuiChip-root"
    onKeyDown={[Function]}
    onKeyUp={[Function]}
  >
    <span
      className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
    >
      Sports
    </span>
  </div>
  <div
    className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1dzpuen-MuiChip-root"
    onKeyDown={[Function]}
    onKeyUp={[Function]}
  >
    <span
      className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
    >
      Universe
    </span>
  </div>
  <div
    className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1dzpuen-MuiChip-root"
    onKeyDown={[Function]}
    onKeyUp={[Function]}
  >
    <span
      className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
    >
      Science
    </span>
  </div>
</div>
`;
